### PR TITLE
#200 switch new run to static atomic

### LINKED ISSUE #200: [Scylla] - Switch new run to static Atomic
### Description

Right now we use a MPSC tokio channel for sending/recving the new run attribute.

### Acceptance Criteria

- Use an atomic instead
- Investigate any other scenarios when an atomic makes more sense

### Proposed Solution

Reduce code complexity and overhead with an `AtomicBool`, using `relaxed` ordering.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by bracyw**
## Changes

Implemented a global AtomicI32 to keep track of current run number.

## Notes

Had to do a global Atomic because Atomics couldn't be passed through a post (doesn't have clone trait).

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `package-lock.json` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #200

### CODE REVIEW

File: scylla-server/src/processors/mqtt_processor.rs [hunk 1]
```diff
@@ -68,14 +62,13 @@ pub struct MqttProcessorOptions {
 impl MqttProcessor {
     /// Creates a new mqtt receiver and socketio and db sender
     /// * `channel` - The mpsc channel to send the database data to
-    /// * `new_run_channel` - The channel for new run notifications
+    /// * `new_run_id` - The changed atomic run id.
```
Comment (jr1221) [inline scylla-server/src/processors/mqtt_processor.rs]: Remove this?

**Review by jr1221 — DISMISSED** [review]
Please just test changing run in client still works.  You can do this by pointing client to scylla, and then incrementing the run in the client and see if it works.

**Comment by RChandler234** [comment]
pls verify that getCurrentRun/ add new run button works correctly in this PR. Right now it's not working on my local (I think bc of an empty db) and causes the UI to say it's disconnected from the backend

File: scylla-server/src/controllers/run_controller.rs [hunk 2]
```diff
@@ -38,16 +41,10 @@ pub async fn get_run_by_id(
...
-    Extension(channel): Extension<mpsc::Sender<run_service::public_run::Data>>,
-) -> Result<Json<PublicRun>, ScyllaError> {
+pub async fn new_run(State(db): State<Database>) -> Result<Json<PublicRun>, ScyllaError> {
     let run_data = run_service::create_run(&db, chrono::offset::Utc::now()).await?;
 
-    // notify the mqtt receiver a new run has been created
-    if let Err(err) = channel.send(run_data.clone()).await {
-        warn!("Could not notify system about an updated run: {}", err);
-    }
+    crate::RUN_ID.store(run_data.id, Ordering::Relaxed);
```
Comment (jr1221) [inline scylla-server/src/controllers/run_controller.rs:44]: can you add a `info` level print notifying about a new run and printing its id?

**Review by jr1221 — DISMISSED** [review]
one thing

**Review by jr1221 — APPROVED** [review]
lgtm

**Pull request merged**